test/QueryPlannerTest.cpp

```diff
@@ -2523,12 +2524,198 @@ TEST(QueryPlanner, SpatialJoinIncorrectConfigValues) {
                 "spatialSearch:maxDistance 5 ;"
                 "spatialSearch:numNearestNeighbors 5 ;"
                 "spatialSearch:algorithm spatialSearch:s2 ;"
-                "spatialSearch:joinType <within-dist> ;"
+                "spatialSearch:joinType <within-dist> ."
                 " { ?a <p> ?b . }"
                 "}}",
                 ::testing::_),
       ::testing::HasSubstr(
           "The selected algorithm does not support the `<joinType>` option"));
+
+  AD_EXPECT_THROW_WITH_MESSAGE(
+      h::expect("PREFIX spatialSearch: "
+                "<https://qlever.cs.uni-freiburg.de/spatialSearch/>"
+                "SELECT * WHERE {"
+                "?x <p> ?y ."
+                "SERVICE spatialSearch: {"
+                "_:config spatialSearch:right ?b ;"
+                "spatialSearch:left ?y ;"
+                "spatialSearch:maxDistance 5 ;"
+                "spatialSearch:algorithm spatialSearch:s2 ;"
+                "spatialSearch:experimentalRightCacheName \"dummy\" . "
+                "}}",
+                ::testing::_),
+      ::testing::HasSubstr(
```

Comment:
We can consider factoring these long tests into their own test files (which then reuse the infrastructure of the queryplannertesthelpers.